### PR TITLE
refactor: replace typography with text component in review recovery phrase

### PR DESCRIPTION
**Pull request by Devin-Apps**
## **Description**

Migrated the `Typography` component to the `Text` component in the `review-recovery-phrase.js` file. This change is part of the ongoing effort to update the codebase to use the new `Text` component for better consistency and maintainability.

Devin Run Link: https://preview.devin.ai/devin/f2332a51ed044002a120b3867453f270

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25265?quickstart=1)

## **Related issues**

Partially Fixes: https://github.com/MetaMask/metamask-extension/issues/17670

## **Manual testing steps**

1. Go to the latest build of storybook in this PR
2. Verify that the `ReviewRecoveryPhrase` component renders correctly with the new `Text` component
3. Ensure that the styling and functionality remain consistent with the previous implementation

### **After**
after

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by Devin-Apps** [comment]
Tagging @georgewrmarshall for review

File: ui/pages/onboarding-flow/recovery-phrase/review-recovery-phrase.js [hunk 1]
```diff
@@ -3,16 +3,16 @@ import { useHistory, useLocation } from 'react-router-dom';
 import PropTypes from 'prop-types';
 import Box from '../../../components/ui/box';
```
Comment (georgewrmarshall) [inline ui/pages/onboarding-flow/recovery-phrase/review-recovery-phrase.js:4]: We could remove this deprecated `Box` and import the component-library verion. The props should all work as we are using enums now
```suggestion
```

File: ui/pages/onboarding-flow/recovery-phrase/review-recovery-phrase.js [hunk 2]
```diff
@@ -3,16 +3,16 @@ import { useHistory, useLocation } from 'react-router-dom';
...
 import Button from '../../../components/ui/button';
-import Typography from '../../../components/ui/typography';
 import { useCopyToClipboard } from '../../../hooks/useCopyToClipboard';
 import { useI18nContext } from '../../../hooks/useI18nContext';
 import { ONBOARDING_CONFIRM_SRP_ROUTE } from '../../../helpers/constants/routes';
+import { Text, Icon, IconName } from '../../../components/component-library';
```
Comment (georgewrmarshall) [inline ui/pages/onboarding-flow/recovery-phrase/review-recovery-phrase.js:9]: Import `Box` here
```suggestion
import {
  Text,
  Icon,
  IconName,
  Box,
} from '../../../components/component-library';
```

**Review by georgewrmarshall — COMMENTED** [review]
Looking good! Left one suggestion

File: ui/pages/onboarding-flow/recovery-phrase/review-recovery-phrase.js (hunk 1 above)
Comment (georgewrmarshall) [reply ui/pages/onboarding-flow/recovery-phrase/review-recovery-phrase.js]: Resolved

**Review by georgewrmarshall — APPROVED** [review]
LGTM! 
- checked storybook ✅ 
- checked branch locally ✅ 

Screenshot 2024-07-23 at 3 46 03 PM

**Review by garrettbear — APPROVED** [review]

**Pull request merged**